flocker/node/agents/gce.py

```diff
@@ -402,19 +413,48 @@ def allocation_unit(self):
 
     def list_volumes(self):
         # TODO(mewert) Walk the pages.
-        result = self._compute.disks().list(project=self._project,
-                                            zone=self._zone).execute()
-        return list(
-            BlockDeviceVolume(
-                blockdevice_id=unicode(disk['name']),
-                size=int(GiB(int(disk['sizeGb'])).to_Byte()),
-                attached_to=_extract_attached_to(disk),
-                dataset_id=_blockdevice_id_to_dataset_id(disk['name'])
+        with start_action(
+                action_type=u"flocker:node:agents:gce:list_volumes",
+        ) as action:
+            result = self._compute.disks().list(project=self._project,
+                                                zone=self._zone).execute()
+
+            # 'description' will not even be in the dictionary if no
+            # description was specified.
+            def disk_in_cluster(disk):
+                return (
+                    disk['name'].startswith(_PREFIX) and
+                    (disk.get('description') ==
```

Comment:
I pose it should not be the case that our code should blow up based on someone outside of flocker rudely deciding to spin up a disk with a malicious name and no description.

I'll add logging to indicate that something fishy is going on in that case, but raising exceptions based on the state of someone's GCE project seems like a bad idea.